code/controllers/subsystem/radiation.dm

```diff
@@ -1,18 +1,117 @@
-PROCESSING_SUBSYSTEM_DEF(radiation)
+SUBSYSTEM_DEF(radiation)
 	name = "Radiation"
-	flags = SS_NO_INIT | SS_BACKGROUND
-	wait = 1 SECONDS
-
-	var/list/warned_atoms = list()
-
-/datum/controller/subsystem/processing/radiation/proc/warn(datum/component/radioactive/contamination)
-	if(!contamination || QDELETED(contamination))
-		return
-	var/ref = REF(contamination.parent)
-	if(warned_atoms[ref])
-		return
-	warned_atoms[ref] = TRUE
-	var/atom/master = contamination.parent
-	SSblackbox.record_feedback("tally", "contaminated", 1, master.type)
-	var/msg = "has become contaminated with enough radiation to contaminate other objects. || Source: [contamination.source] || Strength: [contamination.strength]"
-	master.investigate_log(msg, INVESTIGATE_RADIATION)
+	flags = SS_BACKGROUND | SS_NO_INIT
+
+	wait = 0.5 SECONDS
+
+	/// A list of radiation sources (/datum/radiation_pulse_information) that have yet to process.
+	/// Do not interact with this directly, use `radiation_pulse` instead.
+	var/list/datum/radiation_pulse_information/processing = list()
+
+/datum/controller/subsystem/radiation/fire(resumed)
+	while (processing.len)
+		var/datum/radiation_pulse_information/pulse_information = popleft(processing)
+
+		var/datum/weakref/source_ref = pulse_information.source_ref
+		var/atom/source = source_ref.resolve()
+		if (isnull(source))
+			continue
+
+		pulse(source, pulse_information)
+
+		if (MC_TICK_CHECK)
+			return
+
+/datum/controller/subsystem/radiation/stat_entry(msg)
+	return ..(msg + " | Pulses: [processing.len]")
```

Comment:
```suggestion
/datum/controller/subsystem/radiation/stat_entry(msg)
	msg = "[msg] | Pulses: [processing.len]"
	return ..()
```
This causes less headaches when parameter order changes, but you might have reasons to do it differently.

Reply:
I don't like these because it feels like a gotcha, but I don't have enough objections